src/study/system-model/component.cpp

```diff
@@ -64,6 +66,19 @@ Component::Component(const ComponentData& component_data)
     data_ = std::move(component_data);
 }
 
+const PortFieldNode* Component::getPortFieldNode(std::string portFieldId) const
+{
+    Node* nodeOfPortField = getModel()
+                              ->PortFieldDefinitions()
+                              .at(portFieldId)
+                              .Definition()
+                              .RootNode();
+    // We know for sure that the following dynamic_cast is not nullptr: checks were made
+    // when building port filed definitions.
+    const PortFieldNode* portFieldNode = dynamic_cast<PortFieldNode*>(nodeOfPortField);
```

Comment:
why this cast? definition could be anything, such a variable, parameter or complexe expression. the return type is just Node*

Reply:
Misunderstanding from me.
Corrected now.